src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -737,8 +738,10 @@ Real SpatialPooler::avgConnectedSpanForColumnND_(UInt column) const {
 
 void SpatialPooler::adaptSynapses_(const SDR &input,
                                    const SDR &active) {
+  vector<Permanence> updates( connections_.synapseFlatListLength() , 0.0f );
   for(const auto &column : active.getSparse()) {
-    connections_.adaptSegment(column, input, synPermActiveInc_, synPermInactiveDec_);
+    connections_.adaptSegment(column, input, synPermActiveInc_, synPermInactiveDec_,
+                              updates, updates);
```

Comment:
add check to adaptSegment for updates' size, see the comment push_back vs [i]=x;